pkg/controllers/automigration/controller.go

```diff
@@ -438,3 +494,145 @@ func (c *Controller) getPodsFromCluster(
+				"Failed to get target object form pod",
+				"found", found,
+				"err", err,
+			)
+			continue
+		}
+		managed := object.GetLabels()[managedlabel.ManagedByKubeAdmiralLabelKey] == managedlabel.ManagedByKubeAdmiralLabelValue
+		gkMatched := object.GroupVersionKind().GroupKind() == gvk.GroupKind()
+		if !managed || !gkMatched {
+			c.logger.V(3).Info(
+				"The GVK of Target object not matched",
+				"got-gvk", object.GroupVersionKind(),
+				"managed", managed,
+				"resource", common.NewQualifiedName(object),
+			)
+			continue
+		}
+		possibleQualifies = append(possibleQualifies, common.QualifiedName{
+			Namespace: object.GetNamespace(),
+			Name:      naming.GenerateFederatedObjectName(object.GetName(), ftc.Name),
+		})
+	}
+	return
+}
+
+func (c *Controller) getTargetObjectsIfAutoMigrationEnabled(
+	fedObject fedcorev1a1.GenericFederatedObject,
+) (clusterObjs []FederatedObject, ftc *fedcorev1a1.FederatedTypeConfig, unschedulableThreshold *time.Duration, err error) {
+	// PodUnschedulableThresholdAnnotation is set by the scheduler. Its presence determines whether auto migration is enabled.
+	if value, exists := fedObject.GetAnnotations()[common.PodUnschedulableThresholdAnnotation]; exists {
+		if duration, err := time.ParseDuration(value); err != nil {
+			err = fmt.Errorf("failed to parse PodUnschedulableThresholdAnnotation: %w", err)
+			return nil, nil, nil, err
+		} else {
+			unschedulableThreshold = &duration
+		}
+	}
+
+	objectMeta := &metav1.PartialObjectMetadata{}
+	if err = json.Unmarshal(fedObject.GetSpec().Template.Raw, objectMeta); err != nil {
+		err = fmt.Errorf("failed to unmarshall template of federated object: %w", err)
+		return nil, nil, nil, err
+	}
+	gvk := objectMeta.GroupVersionKind()
+
+	ftc = c.getFTCIfAutoMigrationIsEnabled(gvk)
+	if ftc == nil {
+		return nil, nil, nil, nil
+	}
+
+	for _, placement := range fedObject.GetSpec().Placements {
+		for _, cluster := range placement.Placement {
+			lister, synced, exist := c.federatedInformer.GetResourceLister(gvk, cluster.Cluster)
+			if !exist || !synced() {
+				err = fmt.Errorf("informer of resource %v not exists or not synced for cluster %s", gvk, cluster.Cluster)
+				return nil, nil, nil, err
+			}
+			object, err := lister.ByNamespace(objectMeta.Namespace).Get(objectMeta.Name)
+			if err != nil && !apierrors.IsNotFound(err) {
+				err = fmt.Errorf("failed to get %v from informer stores for cluster %s: %w", objectMeta, cluster.Cluster, err)
+				return nil, nil, nil, err
+			}
+			if apierrors.IsNotFound(err) {
+				continue
+			}
+			clusterObjs = append(clusterObjs, FederatedObject{Object: object, ClusterName: cluster.Cluster})
+		}
+	}
+	return clusterObjs, ftc, unschedulableThreshold, nil
+}
+
+func (c *Controller) setPodInformer(cluster string) {
+	ctx := klog.NewContext(context.TODO(), c.logger)
+	keyedLogger := c.logger.WithValues("cluster", cluster)
+
+	informer, _, exist := c.federatedInformer.GetPodInformer(cluster)
+	if !exist {
```

Comment:
It is possible that the informer does not exist for a new cluster because it has not been processed by the `FederatedInformerManager` yet. What happens in this case? Will the event handlers simply not be set for that cluster indefinitely? If that is the case we fail to process pod changes in that cluster.

Reply:
Perhaps we can delegate the registering of the event handler to the `FederatedInformerManager` when it processes a cluster by exposing a `AddPodEventHander` method on the `FederatedInformerManager` interface.